src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/AnnotationFilterPass.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\DependencyInjection\Compiler;
+
+use ApiPlatform\Core\Api\AnnotatedFilterInterface;
+use ApiPlatform\Core\Util\AnnotationFilterExtractorTrait;
+use ApiPlatform\Core\Util\ReflectionClassRecursiveIterator;
+use Doctrine\Common\Annotations\Reader;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * Injects filters.
+ *
+ * @internal
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class AnnotationFilterPass implements CompilerPassInterface
+{
+    use AnnotationFilterExtractorTrait;
+
+    const TAG_FILTER_NAME = 'api_platform.filter';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $resourceClassDirectories = $container->getParameter('api_platform.resource_class_directories');
+        $reader = $container->get('annotation_reader');
+        $includedFiles = [];
+
+        foreach (ReflectionClassRecursiveIterator::getReflectionClassesFromDirectories($resourceClassDirectories) as $className => $reflectionClass) {
+            $this->createFilterDefinitions($reflectionClass, $reader, $container);
+        }
+    }
+
+    private function createFilterDefinitions(\ReflectionClass $reflectionClass, Reader $reader, ContainerBuilder $container)
+    {
+        foreach ($this->readFilterAnnotations($reflectionClass, $reader) as $id => list($arguments, $filterClass)) {
+            if (!is_subclass_of($filterClass, AnnotatedFilterInterface::class)) {
+                trigger_error(sprintf('The filter "%s" does not implement the "%s" interface. Therefore, we can not create a filter service for this annotation.', $filterClass, AnnotatedFilterInterface::class));
```

Comment:
Seems cleaner then throwing because this is not blocking. Do you have suggestions?